### ISSUE TITLE
[java] Feature request for Law Of Demeter (allow to exclude namespaces)

### ISSUE THREAD

**Issue by alixwar** [pmd/pmd#1945]
**Affects PMD Version:** 
6.10.0

**Rule:**
pmd:LawOfDemeter

**Description:**
When third party (or native) API is designed in a way where it is impossible to follow the Law of Demeter it would be good if such violations could be ignored. I suggest adding a parameter to the rule where one can specify package namespaces to be excluded.

**Suggestions**
Either:
1. Allow to specify a parameter "excludeNamespaces" where one could specify external API package wildcards
2. Allow to specify a parameter "excludeNamespaceDepth" where 1 means that a violating method call from com.test.Foo to com.test2.Bar would trigger a violation (since com == com) but where 2 (com.test != com.test2) would not trigger a violation 

**Running PMD through:** *SonarQube*

**Comment by MalikKillian** [comment]
I would much prefer that PMD would automatically disqualify classes that are not part of the current project. That would clear up probably 90% of the cases where a violation is reported but there's essentially no way to change the behavior.

So an example of possible violations:
```java
public class Foo {
    /**
     * This example will result in two violations.
     */
    public void example(Bar b) {
        // this method call is ok, as b is a parameter of "example"
        C c = b.getC();

        // this method call is a violation, as we are using c, which we got from B.
        // We should ask b directly instead, e.g. "b.doItOnC();"
        c.doIt();

        // this is also a violation, just expressed differently as a method chain without temporary variables.
        b.getC().doIt();

        // a constructor call, not a method call.
        D d = new D();
        // this method call is ok, because we have create the new instance of D locally.
        d.doSomethingElse();
        // this is not a violation, due solely to the fact that class D is not defined within the current
        // project and its behavior cannot reasonably be modified to conform to this rule.
        d.getF().doTheThing();
    }
}
```

For the other "10%" of cases I think an "includeNamespaces" option would actually be more useful. While I might not have the ability (or at least the time and resources) to modify something from `com.apache.*` I might be able to reasonable change packages from `com.acme.*` (i.e. if that library was created within my organization).
